File: nijesmik/week10/단어_변환.js
function solution(begin, target, words) {
  if (!words.includes(target)) {
    return 0;
  }

  let answer = 51;
  const visited = Array(words.length).fill(false);

  const dfs = (begin, depth) => {
    if (depth > answer) {
      return;
    }
    if (begin === target) {
      answer = depth;
    }
    words.forEach((word, i) => {
      if (!visited[i] && diff(begin, word) === 1) {
        visited[i] = true;
        dfs(word, depth + 1);
        visited[i] = false;
      }
    });
  };

  dfs(begin, 0);
  return answer;
}

const diff = (str1, str2) => {
  return [...str1].filter((char, i) => char != str2[i]).length;
};

File: Minah7/week10/단어 변환.java
// 미완성

class Solution {
    static int slen;
    static int wlen;
    static int answer = Integer.MAX_VALUE;
    
    public int solution(String begin, String target, String[] words) {
        slen = begin.length();
        int diff = 0;
        boolean[] d = new int[slen];
        for(int i = 0; i < wlen; i++) {
            if(begin.charAt(i) != target.charAt(i)) {
                diff++;
                d[i] = true;
}
}
        wlen = words.length;
        int cnt = 0;
        ArrayList<Integer>[] arr = new ArrayList[wlen];
        for(int i ) {
}
        dfs(0);
        return answer;
    }
    
    public static void dfs(int start) {
        for(int i = start; i < wlen; i++) {
            
}
}
}